frontend/server/src/Controllers/ACL.php

```diff
@@ -62,4 +62,78 @@ public static function removeGroup(
             'role_id' => $roleId,
         ]));
     }
+
+    /**
+     * Returns all ACLs owned by the current user along with assigned roles for each.
+     *
+     * @param \OmegaUp\Request $r The request object containing user session information.
+     * @throws \OmegaUp\Exceptions\NotFoundException If the user is not found.
+     * @return array{acls: list<array{acl_id: int, type: string, alias: string, users: list<array{user_id: int, username: string, role_id: int, role_name: string, role_description: string}>}>}
+     */
+    public static function apiUserOwnedAclReport(\OmegaUp\Request $r): array {
+        $r->ensureMainUserIdentity();
+
+        $userId = $r->user->user_id;
+
+        // Get ACLs owned by the user with alias, type, and users initialized
+        $ownedAcls = \OmegaUp\DAO\ACLs::getUserOwnedAclTypesWithAliases(
+            $userId
+        );
+
+        if (empty($ownedAcls)) {
+            return ['acls' => []];
+        }
+
+        $aclMap = array_column($ownedAcls, null, 'acl_id');
+        $aclIds = array_keys($aclMap);
+
+        // Build role map (role_id => [name, description])
+        $roleMap = [];
+        foreach (\OmegaUp\DAO\Roles::getAll() as $role) {
+            if (
+                is_null(
+                    $role->role_id
+                ) || is_null(
+                    $role->name
+                ) || is_null(
+                    $role->description
+                )
+            ) {
+                continue;
+            }
+            $roleMap[$role->role_id] = [
+                'name' => $role->name,
+                'description' => $role->description,
+            ];
+        }
+
+        // Get all user roles for these ACLs
+        $userRoles = \OmegaUp\DAO\UserRoles::getByAclIds($aclIds);
+        $userIds = array_filter(
+            array_column(
+                $userRoles,
+                'user_id'
+            ),
+            fn($id) => !is_null($id) // Changed !== null to !is_null()
+        );
+        /** @var list<int> $userIds */
+        $userIds = array_map('intval', array_values(array_unique($userIds)));
```

Comment:
What happens if you avoid parsing to integer?
```suggestion
        $userIds = array_values(array_unique($userIds));
```

At this point, we already know that the `$userIds` elements are of type `int`.

Reply:
updated in lastest commit